go-controller/pkg/testing/mock_chassis.go

```diff
@@ -0,0 +1,65 @@
+package testing
+
+import (
+	"fmt"
+
+	goovn "github.com/ebay/go-ovn"
+	"k8s.io/klog"
+)
+
+// Get logical switch port by name
+func (mock *MockOVNClient) ChassisList() ([]*goovn.Chassis, error) {
+	var chassisCache MockObjectCacheByName
+	var ok bool
+
+	chArray := []*goovn.Chassis{}
+	if chassisCache, ok = mock.cache[ChassisType]; !ok {
+		klog.V(5).Infof("Cache doesn't have any object of type %s", ChassisType)
+		return nil, goovn.ErrorSchema
+	}
+	var ch interface{}
+	for _, ch = range chassisCache {
+		chassis, ok := ch.(*goovn.Chassis)
+		if !ok {
+			return nil, fmt.Errorf("invalid object type assertion for %s", ChassisType)
+		}
+		chArray = append(chArray, chassis)
+	}
+	return chArray, nil
+}
+
+// Delete chassis with given name
+func (mock *MockOVNClient) ChassisDel(chName string) (*goovn.OvnCommand, error) {
+	klog.V(5).Infof("Deleting chassis %s", chName)
+	return &goovn.OvnCommand{
+		Exe: &MockExecution{
+			handler: mock,
+			op:      OpDelete,
+			table:   ChassisType,
+			objName: chName,
+		},
+	}, nil
+}
+
+// Get chassis by hostname or name
+func (mock *MockOVNClient) ChassisGet(name string) ([]*goovn.Chassis, error) {
+	var chassisCache MockObjectCacheByName
+	var ok bool
+	if chassisCache, ok = mock.cache[ChassisType]; !ok {
+		klog.V(5).Infof("Cache doesn't have any object of type %s", ChassisType)
+		return nil, goovn.ErrorSchema
```

Comment:
consider: panic here since this should never happen now. Given https://github.com/ovn-org/ovn-kubernetes/pull/1617/files#r472504580, we shouldn't return `goovn.ErrorSchema` here.

Reply:
I am not sure. Logically it is correct if the type is not defined in the cache, it should return goovn.ErrorSchema. Also, this is a utility library function, isn't it right to return error instead of panic?

I don't have strong opinions about this though.

Reply:
So the only reason I suggested to panic is because now when we create the mock client, we specifically populate the types in the cache. So it will be something truly amiss to not find that type in the cache. The mock_lsp code was returning `goovn.ErrorSchema` earlier because we weren't adding the types to the cache on initialization earlier. I am OK with keeping this as is. But in the past when we have had such `this should never happen` kind of errors in unit-test code, the recommendation has been to panic early.

Reply:
@abhat thanks for the explaination. I will keep it as is since you are ok with it. thanks again for all your time!